lib/spack/llnl/util/filesystem.py

```diff
@@ -282,6 +284,60 @@ def working_dir(dirname, **kwargs):
 
 
 @contextmanager
+def move_transaction(directory_name, parent=None):
+    """Moves a directory to a temporary space. If the operations executed
+    within the context manager don't raise deletes it, otherwise undoes
+    the move operation.
+
+    Args:
+        directory_name (path): absolute path of the directory name
+        parent (path): absolute path of the parent directory where to create
```

Comment:
I'd prefer the name `tmp_parent`. Not a big deal since you define it here but every time I see `parent` I have to remind myself it's not the parent of `directory_name`.

Also I don't see the `parent` argument being set anywhere, including the tests. Is there a reason you foresee this being useful?

Reply:
I agree `parent` may be misleading. I'll go with `tmp_root` if you have no objections.

> Also I don't see the parent argument being set anywhere, including the tests. Is there a reason you foresee this being useful?

This context manager moves a directory into a temporary stage. From there it is either deleted (transaction succeeded) or moved back into its original place (transaction failed). Giving users of this context manager the possibility to specify where the temporary stage is located seemed to me useful / relevant enough to forward the `dir` argument of `tempfile.mkdtemp` upstream.